common/sampling.cpp

```diff
@@ -190,6 +191,7 @@ llama_token llama_sampling_sample(
             llama_sample_tail_free(ctx_main, &cur_p, tfs_z,     min_keep);
             llama_sample_typical  (ctx_main, &cur_p, typical_p, min_keep);
             llama_sample_top_p    (ctx_main, &cur_p, top_p,     min_keep);
+            llama_sample_min_p    (ctx_main, &cur_p, min_p,     min_keep);
```

Comment:
Notice that min_p comes after all samplers, meaning that sampling would be among `top_k->tfs->typical->top_p` results, or, as we have enabled almost every sampler, among `top_k->top_p` results. So, it is among Top K=40, among them Top P 95%, and only then Min P

Reply:
That order seems logical. Min-P is supposed to be a Top-P replacement, but if you were to use it together with other samplers it would fulfill the same role as Top-P and should have the same order. And the default being 0 makes sense since Top-P already has a default value, you don't want to combine both actively at the same time.

Reply:
I agree, it probably shouldn't just be enabled by default. I don't think it was like that when I looked at it, but maybe I missed it.

> why we enable by default almost all samplers? If someone want to use just one sampler, the person have to disable all samplers and penalties

Maybe add a `--disable-all-samplers` option or something? I think the default settings are aimed at generally producing decent results, probably not the case with all samplers disabled.

Reply:
> you don't want to combine both actively at the same time.

Why not? To my understanding, it wouldn't be misleading to apply a Min-P of the top logits that combined make a certain procentage since Min-P takes into consideration the procentage of the first (highest probability) logit.

Reply:
Yes, it does seem like the default is currently 0.05, which probably should be changed to 0.